File: CS/GridControlWithResizingColumnHeader/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace GridControlWithResizingColumnHeader
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            Random rnd = new Random();
            DataTable table = new DataTable();
            table.Columns.Add("Value1");
            table.Columns.Add("Value2");
            for (int i = 0; i < rnd.Next(100); i++)
            {
                table.Rows.Add(new object[] { "Custom" + rnd.Next(15), "Address" + rnd.Next(20) });
            }
            myGridControl1.DataSource = table;
        }
    }
}

File: VB/GridControlWithResizingColumnHeader/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace GridControlWithResizingColumnHeader
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            Dim rnd As New Random()
            Dim table As New DataTable()
            table.Columns.Add("Value1")
            table.Columns.Add("Value2")
            Dim i As Integer = 0
            Do While i < rnd.Next(100)
                table.Rows.Add(New Object() { "Custom" & rnd.Next(15), "Address" & rnd.Next(20) })
                i += 1
            Loop
            myGridControl1.DataSource = table
        End Sub
    End Class
End Namespace
